### DISCUSSION TITLE
Could not find prisma-fmt binary

### DISCUSSION THREAD

**Discussion by HitsiLent** [keystonejs/keystone#7461] [Questions] [answered]
Generating GraphQL and Prisma schemas Error: Could not find prisma-fmt binary when you run yarn dev

1. After yarn install version Release - 5th October 2021
2. throw error while run yarn dev on blog example project 

I am using macOS

**Comment by timleslie** [comment]
Hi @HitsiLent, let's make sure you have a clean and correct install. Please run the following commands and see if it fixes things.

```
rm -rf .keystone
rm -rf node_modules
yarn
yarn dev
```

**Comment by HitsiLent** [comment]
same error, i think it cuz upgrade to prisma 3.X cause the problem

**Comment by Zegnat** [comment]
I just ran into the exact same issue just now, with a completely clean install. Putting a copy of the error here as text, rather than just a screenshot. Ran `npm init keystone-app`, then `cd`ed into the new folder, and then `npm run dev`:

```
> keystone-app@3.0.1 dev /Users/zegnat/keystone
> keystone-next dev

✨ Starting Keystone
⭐️ Dev Server Starting on http://localhost:3000
⭐️ GraphQL API Starting on http://localhost:3000/api/graphql
✨ Generating GraphQL and Prisma schemas
Error: Could not find prisma-fmt binary. Searched in /Users/zegnat/keystone/node_modules/@prisma/sdk/prisma-fmt-darwin and /Users/zegnat/keystone/node_modules/@prisma/prisma-fmt-darwin
    at resolveBinary (/Users/zegnat/keystone/node_modules/@prisma/sdk/dist/resolveBinary.js:87:9)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at Object.formatSchema (/Users/zegnat/keystone/node_modules/@prisma/sdk/dist/engine-commands/formatSchema.js:41:25)
    at getCommittedArtifacts (/Users/zegnat/keystone/node_modules/@keystone-next/keystone/artifacts/dist/keystone-next-keystone-artifacts.cjs.dev.js:427:13)
    at Object.generateCommittedArtifacts (/Users/zegnat/keystone/node_modules/@keystone-next/keystone/artifacts/dist/keystone-next-keystone-artifacts.cjs.dev.js:490:21)
    at initKeystone (/Users/zegnat/keystone/node_modules/@keystone-next/keystone/scripts/dist/keystone-next-keystone-scripts.cjs.dev.js:308:27)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! keystone-app@3.0.1 dev: `keystone-next dev`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the keystone-app@3.0.1 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/zegnat/.npm/_logs/2021-11-01T07_49_44_476Z-debug.log
```

Bit of system info, incase that is important:

```fish
system_profiler SPSoftwareDataType
# Software:
# 
#     System Software Overview:
# 
#       System Version: macOS 11.6 (20G165)
#       Kernel Version: Darwin 20.6.0
#       Boot Volume: Macintosh HD
#       Boot Mode: Normal
#       Computer Name: Zegnat’s MacBook Air
#       User Name: Martijn (zegnat)
#       Secure Virtual Memory: Enabled
#       System Integrity Protection: Enabled
#       Time since boot: 25 days 3:07

node --version
# v14.17.6

npm --version
# 6.14.15
```

**Comment by Zegnat** [comment]
My own comment actually showed me the way. Switching to the newer LTS Node (v16) made it run correctly. I am guessing there is some soft requirement between depencendies and node version and what not. @HitsiLent what version of node/yarn were you using?

**Comment by sladg** [comment]
For me, the issue was including `--ignore-scripts` in my Docker file during `yarn install`.

Following image works for me well.
```
# This is using node 16.13
FROM node:gallium-bullseye-slim
```

**Comment by vonba** [comment]
I am seeing the same issue after upgrading to keystone-6 from keystone-next
Using Node 16.13

**Comment by remygendron** [comment]
Same issue when deploying KS 6 to AWS Elastic Beanstalk. Runs fine on Node 14 on MacOS, but fails with the following on same Node 14 version on Amazon Linux 2 images under Elastic Beanstalk.

Error: Could not find prisma-fmt binary. Searched in /var/app/current/node_modules/@prisma/sdk/prisma-fmt-rhel-openssl-1.0.x and /var/app/current/node_modules/@prisma/prisma-fmt-rhel-openssl-1.0.x
Feb  9 01:28:52 ip-10-0-69-95 web: at resolveBinary (/var/app/current/node_modules/@prisma/sdk/dist/resolveBinary.js:92:9)
Feb  9 01:28:52 ip-10-0-69-95 web: at Object.formatSchema (/var/app/current/node_modules/@prisma/sdk/dist/engine-commands/formatSchema.js:41:25)
Feb  9 01:28:52 ip-10-0-69-95 web: at getCommittedArtifacts (/var/app/current/node_modules/@keystone-6/core/dist/artifacts-7f46986c.cjs.dev.js:398:13)
Feb  9 01:28:52 ip-10-0-69-95 web: at Object.generateCommittedArtifacts (/var/app/current/node_modules/@keystone-6/core/dist/artifacts-7f46986c.cjs.dev.js:461:21)
Feb  9 01:28:52 ip-10-0-69-95 web: at setupInitialKeystone (/var/app/current/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.dev.js:402:25)
Feb  9 01:28:52 ip-10-0-69-95 web: at initKeystone (/var/app/current/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.dev.js:165:35)

**Comment by remygendron** [comment]
Additional info from my Elastic Beanstalk deployment.

Looked at the code for resolveBinary.js. And looked on the ec2 instance where that code is looking for the binary. The paths and binary are simply not there. Looked at all the eb logs at deployment time, no visible errors when installing npm dependencies. 

On my mac, the binary is at : node_modules/@prisma/engines/prisma-fmt-darwin

On the ec2 instance, prisma-fmt-* cannot be found under node_modules/@prisma with find . -name "prisma-fmt*"

I conclude that the binary was simply not built on the AWS Linux 2 AMI.

**Comment by remygendron** [comment]
Other ideas:

As AWS EB installs a Node app in production mode (npm install --production), the package-lock.jon file generated on my Mac may incite npm install to consider installing some MacOS version of packages. Not sure about that, but just in case.

I rm -rf the node_modules folder, and deleted package-lock.json and did a plain npm install.

Same error at launch.

Running out of ideas.

**Comment by molomby** [comment, marked as answer]
Digging though the `prisma@3.6.0` package, it depends on `@prisma/engines@3.6.0-24.dc520b92b1ebb2d28dc3161f9f82e875bd35d727`. That package has a `postinstall` script that downloads the appropriate engine binary.

Could your build environment be blocking this request? This might explain why you have the package code but are missing the binary. You must be allowing requests to `npmjs.org` (or an internal mirror?) otherwise you wouldn't be able to install the package itself. The Prisma binary comes from `binaries.prisma.sh`, so could be separately be being blocked. Or maybe HTTP requests aren't blocked but your build env can't resolve that DNS for some reason..?

Looking at https://binaries.prisma.sh, it's a publicly listable S3 bucket. You could probably manually pull the binaries you need and incorporate them into your project some how if that better fitted your use case.

**Comment by remygendron** [comment]
Thank you for the repry molomby. The ec2 instance runs on a public subnet and I verified that it has a security group that allows all traffic outbound. This is standard and is a sg that is configured by Elastic Beanstalk at deploy time. The IAM role also has AdministratorAccess.

I would building a deploy recipe for a low code backend and wanted to keep things simple. As such, I was deploying directly to an EB Node Platform. I am in the process off containerizing the deployment and will use EB Docker platform. This will also allow me to upgrade to Node 16 (eb only support node 14 natively).

Thank you for your input.

**Comment by robwilkerson** [comment]
Just bumped into this issue trying to install keystone in a docker container based on `node:16-alpine`. I'm doing this locally, so access shouldn't be an issue. Going to keep searching, but thought I'd add my voice here.

**Reply by dcousens** [reply]
Are you using the yarn flag `--ignore-scripts`?

**Comment by mwcz** [comment]
I'm also running into this error.  Not in a build environment, but on my regular workstation.  DNS for `binaries.prisma.sh` resolves.  I've tried with node 14, 16, and 17, with yarn 1.22 and 3.2, and with npm versions 6.14, 8.5, and 8.5.5.  I'm on Linux (Fedora 36).

<details>
<summary>yarn install  log</summary>
<pre><code>
$ yarn
yarn install v1.22.17
info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
warning @keystone-6/core > @prisma/sdk > temp-write > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
warning @keystone-6/core > supertest > superagent@7.1.2: Deprecated due to bug in CI build https://github.com/visionmedia/superagent/pull/1677\#issuecomment-1081361876
warning @keystone-6/fields-document > @braintree/sanitize-url@5.0.2: Potential XSS vulnerability patched in v6.0.0.
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning " > @keystone-6/auth@1.0.2" has unmet peer dependency "react@^17.0.2".
warning "@keystone-6/auth > @keystone-ui/core@4.0.0" has unmet peer dependency "react@^17.0.2".
warning "@keystone-6/auth > @keystone-ui/core@4.0.0" has unmet peer dependency "react-dom@^17.0.2".
warning "@keystone-6/auth > @keystone-ui/core > @emotion/react@11.8.2" has unmet peer dependency "react@>=16.8.0".
warning "@keystone-6/auth > @keystone-ui/core > @emotion/react > @emotion/babel-plugin@11.7.2" has unmet peer dependency "@babel/core@^7.0.0".
warning "@keystone-6/auth > @keystone-ui/core > @emotion/react > @emotion/babel-plugin > @babel/plugin-syntax-jsx@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@keystone-6/auth > @keystone-ui/fields > react-select@5.2.2" has unmet peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@keystone-6/core > @prisma/migrate@3.11.0" has unmet peer dependency "@prisma/generator-helper@*".
warning "@keystone-6/fields-document > slate-react@0.69.0" has unmet peer dependency "react-dom@>=16.8.0".
[5/5] Building fresh packages...
success Saved lockfile.
$ keystone postinstall
Error: Could not find prisma-fmt binary. Searched in /home/user/projects/keystone-test/node_modules/@prisma/sdk/prisma-fmt-rhel-openssl-3.0.x and /home/user/projects/keystone-test/node_modules/@prisma/prisma-fmt-rhel-openssl-3.0.x
    at resolveBinary (/home/user/projects/keystone-test/node_modules/@prisma/sdk/dist/resolveBinary.js:92:9)
    at Object.formatSchema (/home/user/projects/keystone-test/node_modules/@prisma/sdk/dist/engine-commands/formatSchema.js:41:25)
    at getCommittedArtifacts (/home/user/projects/keystone-test/node_modules/@keystone-6/core/dist/artifacts-61fd8d1d.cjs.dev.js:398:13)
    at Object.validateCommittedArtifacts (/home/user/projects/keystone-test/node_modules/@keystone-6/core/dist/artifacts-61fd8d1d.cjs.dev.js:417:21)
    at postinstall (/home/user/projects/keystone-test/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.dev.js:619:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
</code></pre>
</details>

**Reply by dcousens** [reply]
Are you using the yarn flag `--ignore-scripts`?

**Reply by mwcz** [reply]
No, in the log I posted there is no `--ignore-scripts`.

**Comment by smarques** [comment]
Same problem here, using Docker with node 16.14.2, also tried gallium-bullseye-slim (no --ignore-scripts)

**Reply by dcousens** [reply]
@smarques this problem happens when the Prisma `postinstall` script has not run, or has failed.  Did you have any other errors?

**Reply by smarques** [reply]
@dcousens not on the docker logs, I will try to replicate this directly from command line to see if I get more info.... It's strange, I have built the same container on my machine about 6 days ago and it worked.. :/. Let me investigate and get back to you, thank you

**Reply by smarques** [reply]
@dcousens here is an update, it works on Linux and MacOs, but the SAME docker image gives me the error on a Window10 machine. I can not figure out how this should happen, I thought the docker env would be deterministic, maybe I am missing something.

**Reply by socialwell-demos** [reply]
In localhost environment, same issue. Had to use FROM node:16. Docker image size is 1.8 GB

**Comment by mikesimps** [comment]
So digging a little deeper on this and the build script is looking in the @prisma/sdk directory, but in my local env it seems to have the binary in the @prisma/engines directory. This is built using the node:16-slim image on a Mac M1 mini.

Looking in `/var/service/node_modules/@prisma/sdk/prisma-fmt-debian-openssl-1.1.x`

![image](https://user-images.githubusercontent.com/3085186/163730107-98b434d5-bf34-4262-9afa-32d4f619fe9d.png)

Actually located in `node_modules/@prisma/engines/prisma-fmt-darwin-arm64` (from my local dev env, not the image)

![image](https://user-images.githubusercontent.com/3085186/163730090-70d984e7-c2b1-4106-837b-2a5fbc615e0a.png)

I realize that my local dev env is going to be different then the target docker image (linux) but I would assume it would put them in the same place. The results are the same building with a linux target or without `--platform=linux/amd64`, only it assumes the `prisma-fmt-linux-arm64-openssl-1.1.x` is in the same location.

I didn't see any way to influence this other than maybe a symlink but that seems fragile. `yarn build` locally runs without issues. Hopefully this helps generate some ideas from more knowledgeable people than myself.

**Comment by sharno** [comment]
I'm getting the same issue too
found out that the binary exists in the engines folder and not in the sdk folder:

```
$ /home/sharno/projects/learn-keystone/node_modules/.bin/keystone dev
✨ Starting Keystone
⭐️ Dev Server Starting on http://localhost:3000
⭐️ GraphQL API Starting on http://localhost:3000/api/graphql
✨ Generating GraphQL and Prisma schemas
Error: Could not find prisma-fmt binary. Searched in /home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/node_modules/@prisma/sdk/prisma-fmt-debian-openssl-3.0.x and /home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/node_modules/@prisma/prisma-fmt-debian-openssl-3.0.x
    at resolveBinary (/home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/node_modules/@prisma/sdk/dist/resolveBinary.js:92:9)
    at Object.formatSchema (/home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/node_modules/@prisma/sdk/dist/engine-commands/formatSchema.js:41:25)
    at getCommittedArtifacts (/home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/dist/artifacts-61fd8d1d.cjs.dev.js:398:13)
    at Object.generateCommittedArtifacts (/home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/dist/artifacts-61fd8d1d.cjs.dev.js:461:21)
    at setupInitialKeystone (/home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.dev.js:403:25)
    at initKeystone (/home/sharno/projects/learn-keystone/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.dev.js:166:35)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Reply by camopy** [reply]
Same here, clean install with node 16

**Reply by dbaker-aa** [reply]
Same here, tried a clean installs with npm & yarn. 

Trying to spin it up on Ubuntu 22.04 with Amazon EC2 - Currently have it running on windows 10 swimmingly

**Reply by dbaker-aa** [reply]
I was digging through some PRs and I found this https://github.com/keystonejs/keystone/pull/7489

Maybe this will fix our issue?

**Reply by dcousens** [reply]
@dbaker-aa to my knowledge https://github.com/prisma/prisma/issues/11356 (fixed by `prisma@3.13.0`) shouldn't affect you, but if that is the problem, that should be resolved in the next release.

**Reply by dbaker-aa** [reply]
Downgraded to Ubuntu 20 from 22 and now things are running swimmingly :)

**Reply by camopy** [reply]
I updated Prisma to the latest version, generated its binary, and placed it in the correct folder for Keystone. But still throwing errors:

```
✨ Starting Keystone
⭐️ Dev Server Starting on http://localhost:5000
⭐️ GraphQL API Starting on http://localhost:5000/api/graphql
✨ Generating GraphQL and Prisma schemas
✨ The database is already in sync with the Prisma schema.
Error: Unknown binary target debian-openssl-3.0.x in generator client.
Possible binaryTargets: darwin, darwin-arm64, debian-openssl-1.0.x, debian-openssl-1.1.x, rhel-openssl-1.0.x, rhel-openssl-1.1.x, linux-arm64-openssl-1.1.x, linux-arm64-openssl-1.0.x, linux-arm-openssl-1.1.x, linux-arm-openssl-1.0.x, linux-musl, linux-nixos, windows, freebsd11, freebsd12, openbsd, netbsd, arm, native
    at validateGenerators (/home/camopy/dev/plur-cms/node_modules/@prisma/sdk/dist/get-generators/getGenerators.js:318:17)
    at getGenerators (/home/camopy/dev/plur-cms/node_modules/@prisma/sdk/dist/get-generators/getGenerators.js:122:3)
    at Object.getGenerator (/home/camopy/dev/plur-cms/node_modules/@prisma/sdk/dist/get-generators/getGenerators.js:276:22)
    at generatePrismaClient (/home/camopy/dev/plur-cms/node_modules/@keystone-6/core/dist/artifacts-f7bed9de.cjs.dev.js:522:21)
    at async Promise.all (index 0)
    at Object.generateNodeModulesArtifacts (/home/camopy/dev/plur-cms/node_modules/@keystone-6/core/dist/artifacts-f7bed9de.cjs.dev.js:518:3)
    at async Promise.all (index 0)
    at setupInitialKeystone (/home/camopy/dev/plur-cms/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.dev.js:416:22)
    at initKeystone (/home/camopy/dev/plur-cms/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.dev.js:166:35)
```

**Reply by dbaker-aa** [reply]
@camopy  what OS are you trying to get this running on? Is there a lower (but stable & supported) version that you can try

**Reply by camopy** [reply]
> @camopy what OS are you trying to get this running on? Is there a lower (but stable & supported) version that you can try

The latest version of pop-os (22.04 LTS)

**Reply by dbaker-aa** [reply]
> > @camopy what OS are you trying to get this running on? Is there a lower (but stable & supported) version that you can try
> 
> The latest version of pop-os (22.04 LTS)

It looks like the latest pop-os might be running on Ubuntu 22.04, which I had a lot of issues running it.   It might be worth a shot trying Pop-os 21.10, but it looks like they are dropping support for it in July of this year.

**Reply by igli7** [reply]
I tried to run on Ubuntu 22.04 LTS and no success.

**Reply by dbaker-aa** [reply]
> I tried to run on Ubuntu 22.04 LTS and no success.

Same - Works great on Ubuntu 20, which is supported for at least the next year.

**Reply by igli7** [reply]
> > I tried to run on Ubuntu 22.04 LTS and no success.
> 
> 
> 
> Same - Works great on Ubuntu 20, which is supported for at least the next year. 

I'm not willing to downgrade to Ubuntu 20, I'm just going to use another cms. It's sad that they haven't fixed this yet.

**Reply by dcousens** [reply]
@Igli7 what errors are you receiving?

**Comment by Alexnortung** [comment]
I am also getting a similar issue.
I had no luck in finding the binary.
This was on a fresh install on my NixOS system.

```
Error: Could not find prisma-fmt binary. Searched in /home/alexander/source/keystone-test/awesome/node_modules/@prisma/sdk/prisma-fmt-linux-nixos and /home/alexander/source/keystone-test/awesome/node_modules/@prisma/prisma-fmt-linux-nixos
```

**Comment by igli7** [comment]
any workaround on this?

**Reply by dcousens** [reply]
Hi @Igli7,
We have merged https://github.com/keystonejs/keystone/pull/7595 which might help when this happens during development

**Reply by igli7** [reply]
I tried to run it on Ubuntu 22.04 LTS and no success.

**Reply by Achisingh** [reply]
Could you try upgrading to `@keystone-6/core@2.0.0` and see if the issue persists?

**Comment by kilinkis** [comment]
experiencing the same issue :(

**Reply by Achisingh** [reply]
Could you try upgrading to `@keystone-6/core@2.0.0` and see if the issue persists?

**Comment by Twoy519** [comment]
Is there a workaround / best practice for how to deal with this error coming up during a heroku build? I'm hitting the following error message when deploying keystone-6 to heroku. 

```
remote:        npm ERR! Error: Could not find prisma-fmt binary. Searched in:
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/engines/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/sdk/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/sdk/runtime/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR!     at resolveBinary (/tmp/build_46c97497/node_modules/@prisma/sdk/dist/resolveBinary.js:91:9)
remote:        npm ERR!     at Object.formatSchema (/tmp/build_46c97497/node_modules/@prisma/sdk/dist/engine-commands/formatSchema.js:41:25)
remote:        npm ERR!     at getCommittedArtifacts (/tmp/build_46c97497/node_modules/@keystone-6/core/dist/artifacts-1f28c573.cjs.prod.js:398:13)
remote:        npm ERR!     at Object.validateCommittedArtifacts (/tmp/build_46c97497/node_modules/@keystone-6/core/dist/artifacts-1f28c573.cjs.prod.js:417:21)
remote:        npm ERR!     at postinstall (/tmp/build_46c97497/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.prod.js:616:5)
```


Full build log:
```
~ git subtree push --prefix keystone_backend heroku main
git push using:  heroku main
Enumerating objects: 31066, done.
Counting objects: 100% (31066/31066), done.
Delta compression using up to 8 threads
Compressing objects: 100% (20848/20848), done.
Writing objects: 100% (31066/31066), 126.93 MiB | 3.44 MiB/s, done.
Total 31066 (delta 8723), reused 29565 (delta 8115), pack-reused 0
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Building on the Heroku-22 stack
remote: -----> Determining which buildpack to use for this app
remote: -----> Node.js app detected
remote:
remote: -----> Creating runtime environment
remote:
remote:        NPM_CONFIG_LOGLEVEL=error
remote:        NODE_VERBOSE=false
remote:        NODE_ENV=production
remote:        NODE_MODULES_CACHE=true
remote:
remote: -----> Installing binaries
remote:        engines.node (package.json):  ^16.16
remote:        engines.npm (package.json):   unspecified (use default)
remote:
remote:        Resolving node version ^16.16...
remote:        Downloading and installing node 16.16.0...
remote:        Using default npm version: 8.11.0
remote:
remote: -----> Installing dependencies
remote:        Prebuild detected (node_modules already exists)
remote:        Rebuilding any native modules
remote:        npm ERR! code 1
remote:        npm ERR! path /tmp/build_46c97497
remote:        npm ERR! command failed
remote:        npm ERR! command sh -c keystone postinstall
remote:        npm ERR! Error: Could not find prisma-fmt binary. Searched in:
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/engines/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/sdk/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR! - /tmp/build_46c97497/node_modules/@prisma/sdk/runtime/prisma-fmt-debian-openssl-3.0.x
remote:        npm ERR!     at resolveBinary (/tmp/build_46c97497/node_modules/@prisma/sdk/dist/resolveBinary.js:91:9)
remote:        npm ERR!     at Object.formatSchema (/tmp/build_46c97497/node_modules/@prisma/sdk/dist/engine-commands/formatSchema.js:41:25)
remote:        npm ERR!     at getCommittedArtifacts (/tmp/build_46c97497/node_modules/@keystone-6/core/dist/artifacts-1f28c573.cjs.prod.js:398:13)
remote:        npm ERR!     at Object.validateCommittedArtifacts (/tmp/build_46c97497/node_modules/@keystone-6/core/dist/artifacts-1f28c573.cjs.prod.js:417:21)
remote:        npm ERR!     at postinstall (/tmp/build_46c97497/node_modules/@keystone-6/core/scripts/dist/keystone-6-core-scripts.cjs.prod.js:616:5)
remote:
remote:        npm ERR! A complete log of this run can be found in:
remote:        npm ERR!     /tmp/npmcache.xSuJg/_logs/2022-08-04T20_15_33_265Z-debug-0.log
remote:
remote: -----> Build failed
remote:
remote:        We're sorry this build is failing! You can troubleshoot common issues here:
remote:        https://devcenter.heroku.com/articles/troubleshooting-node-deploys
remote:
remote:        Some possible problems:
remote:
remote:        - node_modules checked into source control
remote:          https://devcenter.heroku.com/articles/node-best-practices#only-git-the-important-bits
remote:
remote:        Love,
remote:        Heroku
remote:
remote:  !     Push rejected, failed to compile Node.js app.
remote:
remote:  !     Push failed
remote: Verifying deploy...
remote:
remote: !	Push rejected to heroku-keystone-graphql-api.
remote:
To https://git.heroku.com/heroku-keystone-graphql-api.git
 ! [remote rejected]   630c7b8af168edfadcfad5633a4893a014c03425 -> main (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/heroku-keystone-graphql-api.git'
```

Here is my `package.json`:
```
{
  "name": "keystone-app",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "dev": "keystone dev",
    "start": "keystone start",
    "build": "keystone build",
    "postinstall": "keystone postinstall"
  },
  "dependencies": {
    "@keystone-6/auth": "^1.0.1",
    "@keystone-6/core": "^1.0.1",
    "@keystone-6/fields-document": "^1.0.1",
    "dotenv": "^16.0.1",
    "typescript": "^4.5.4"
  },
  "engines": {
    "node": "^16.16"
  }
}
```

I don't have any buildpacks installed in heroku

**Reply by Twoy519** [reply]
`npm run dev` works just fine

**Reply by Twoy519** [reply]
Upgrading to `@keystone-6/core@2.1.0` worked.

fwiw it was painful for me as a n00b to understand the react version dependency chain from react 17 -> 18 between `@keystone-6/core`, `@keystone-6/auth` and `@keystone-6/fields-document`. I ended up just using `npm i @keystone-6/auth --legacy-peer-deps` but I now see a bunch of log spew when I run `npm i` which makes me feel like I'm using a broken system. Is there a better fix for this?

New `package.json`:

```
{
  "name": "my-app",
  "version": "2.0.0",
  "private": true,
  "scripts": {
    "dev": "keystone dev",
    "start": "keystone start",
    "build": "keystone build",
    "postinstall": "keystone postinstall"
  },
  "dependencies": {
    "@keystone-6/auth": "^4.0.0",
    "@keystone-6/core": "^2.1.0",
    "@keystone-6/fields-document": "^4.0.1",
    "dotenv": "^16.0.1",
    "typescript": "^4.5.4"
  },
  "engines": {
    "node": "^16.16"
  }
}
```

**Comment by zazin** [comment]
thanks, its worked for me too after upgrading "@keystone-6/core": "^2.1.0"
